geoips/filenames/base_paths.py

```diff
@@ -1,252 +1,171 @@
-
-PATHS["TC_DECKS_DB"] = pathjoin(
-    PATHS["GEOIPS_OUTDIRS"], "longterm_files", "tc", "tc_decks.db"
-)
-if getenv("TC_DECKS_DB"):
-    PATHS["TC_DECKS_DB"] = getenv("TC_DECKS_DB")
-
-if getenv("TC_DECKS_DIR"):
-    PATHS["TC_DECKS_DIR"] = getenv("TC_DECKS_DIR").rstrip("/")
-else:
-    PATHS["TC_DECKS_DIR"] = pathjoin(
-        PATHS["GEOIPS_OUTDIRS"], "longterm_files", "tc", "decks"
-    )
+    geoips_global_variables = {
+        # GeoIPS Documentation URL
+        "GEOIPS_DOCS_URL": r"https://nrlmmd-geoips.github.io/geoips/",
+        # Version
+        "GEOIPS_VERS": os.getenv("GEOIPS_VERS", "0.0.0"),
+        # Operational User
+        "GEOIPS_OPERATIONAL_USER": os.getenv("GEOIPS_OPERATIONAL_USER", False),
+        # Copyright Information
+        "GEOIPS_COPYRIGHT": os.getenv("GEOIPS_COPYRIGHT", "NRL-Monterey"),
+        "GEOIPS_COPYRIGHT_ABBREVIATED": os.getenv(
+            "GEOIPS_COPYRIGHT_ABBREVIATED", "NRLMRY"
+        ),
+        # Configuration and Queue
+        "GEOIPS_RCFILE": os.getenv("GEOIPS_RCFILE", ""),
+        "DEFAULT_QUEUE": os.getenv("DEFAULT_QUEUE", None),
+        # Computer Identifier
+        "BOXNAME": socket.gethostname(),
+    }
+
+    paths.update(geoips_global_variables)
+
+    derivative_directory_paths = {
+        paths["GEOIPS_BASEDIR"]: {
+            "GEOIPS_TESTDATA_DIR": "test_data",
+            "GEOIPS_DEPENDENCIES_DIR": "geoips_dependencies",
+        },
+        paths["GEOIPS_OUTDIRS"]: {
+            # Data Output Paths
+            "PRESECTORED_DATA_PATH": "preprocessed/sectored",
+            "PREREAD_DATA_PATH": "preprocessed/unsectored",
+            "PREREGISTERED_DATA_PATH": "preprocessed/registered",
+            "PRECALCULATED_DATA_PATH": "preprocessed/algorithms",
+            "CLEAN_IMAGERY_PATH": "preprocessed/clean_imagery",
+            "ANNOTATED_IMAGERY_PATH": "preprocessed/annotated_imagery",
+            "FINAL_DATA_PATH": "preprocessed/final",
+            "PREGENERATED_GEOLOCATION_PATH": "preprocessed/geolocation",
+            # Scratch Directories
+            "SCRATCH": "scratch",
+            "LOCALSCRATCH": "scratch",
+            "SHAREDSCRATCH": "scratch",
+            # Log and Data Directories
+            "LOGDIR": "logs",
+            "GEOIPSDATA": "geoipsdata",
+            # Ancillary Data Directories
+            "GEOIPS_ANCILDAT_AUTOGEN": "ancildat_autogen",
+            "GEOIPS_ANCILDAT": "ancildat",
+            # WWW Paths
+            "TCWWW": "preprocessed/tcwww",
+            "PUBLICWWW": "preprocessed/publicwww",
+            "PRIVATEWWW": "preprocessed/privatewww",
+            # Tropical Cyclone Paths
+            "TC_DECKS_DB": "longterm_files/tc/tc_decks.db",
+            "TC_DECKS_DIR": "longterm_files/tc/decks",
+        },
+        paths["BASE_PATH"]: {
+            "TC_TEMPLATE": "plugins/yaml/sectors/dynamic/tc_web_template.yaml",
+        },
+    }
+
+    for top_directory, sub_directories in derivative_directory_paths.items():
```

Comment:
Leave a comment here that we are looping through all the directory-based paths, and using "get_env_var" function to set the paths to the environment variable specified option when defined, else defaulting to the passed-in default (second argument).  Also note that in the future, this will likely also pull from a YAML or JSON config file (maybe still support env var override?  Not sure).

Reply:
Added! :)